integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataLoadRDD.scala

```diff
@@ -159,6 +166,31 @@ class CarbonDataLoadRDD[K, V](
         }
         carbonLoadModel.setSegmentId(String.valueOf(loadCount))
         setModelAndBlocksInfo()
+        val surrogatekeysTotalTimeArray = new Array[Long](2)
```

Comment:
I did not get what you are trying to do here. It supposed everything abstract. Handle every thing inside your statistics class.